src/utils/filterLearningTheme/filterLearningTheme.ts

```diff
@@ -5,7 +5,26 @@ export const filterLearningTheme = (
   units: UnitListingData["units"]
 ): UnitListingData["units"] => {
   if (themeSlug) {
-    return units.filter((unit) => unit[0]?.themeSlug === themeSlug);
+    const filteredUnits = [];
+
+    for (const unitVariant of units) {
```

Comment:
I wonder if there is a clearer way to write this, rather than using 3 nested for loops. Could use the .some array method rather then having a hasMatch flag?

Reply:
Refactored code ✅